### PR TITLE
Add board definition for Seeed Xiao RP2040

### PR DESCRIPTION
**Pull request by ryanplusplus**
https://www.seeedstudio.com/XIAO-RP2040-v1-0-p-5026.html
https://wiki.seeedstudio.com/XIAO-RP2040/

After posting this I found https://github.com/raspberrypi/pico-sdk/pull/524. This addresses the remaining comments on that PR.

### CODE REVIEW

File: src/boards/include/boards/seeed_xiao_rp2040.h [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#define PICO_DEFAULT_SPI_RX_PIN 4
+#endif
+
+#ifndef PICO_DEFAULT_SPI_SCK_PIN
+#define PICO_DEFAULT_SPI_SCK_PIN 2
+#endif
+
+//------------- FLASH -------------//
+
+#define PICO_BOOT_STAGE2_CHOOSE_GENERIC_03H 1
```
Comment (ryanplusplus) [inline src/boards/include/boards/seeed_xiao_rp2040.h:85]: The schematic gives the flash chip as a GigaDevice GD25Q16CEIG.

Comment (ryanplusplus) [reply src/boards/include/boards/seeed_xiao_rp2040.h:85]: This also works and should be faster:

```suggestion
#define PICO_BOOT_STAGE2_CHOOSE_W25Q080 1
```

File: src/boards/include/boards/seeed_xiao_rp2040.h [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#define PICO_DEFAULT_SPI_RX_PIN 4
+#endif
+
+#ifndef PICO_DEFAULT_SPI_SCK_PIN
+#define PICO_DEFAULT_SPI_SCK_PIN 2
+#endif
+
+//------------- FLASH -------------//
+
+#define PICO_BOOT_STAGE2_CHOOSE_W25Q080 1
```
Comment (ryanplusplus) [inline src/boards/include/boards/seeed_xiao_rp2040.h:85]: The schematic gives the flash chip as a GigaDevice GD25Q16CEIG. As far as I can tell it is compatible with W25Q080.

Comment (mcer12) [reply src/boards/include/boards/seeed_xiao_rp2040.h:85]: @ryanplusplus W25Q080 is 8mb chip, GD25Q16 is 16mb or am I missing something?

Comment (ryanplusplus) [reply src/boards/include/boards/seeed_xiao_rp2040.h:85]: You're right, the size is mismatched. Do you know whether the flash type specified here does more than just select the protocol?

Comment (kilograham) [reply src/boards/include/boards/seeed_xiao_rp2040.h:85]: it just picks the boot2 which sets up the protocol.

File: src/boards/include/boards/seeed_xiao_rp2040.h [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+#ifndef PICO_FLASH_SPI_CLKDIV
+#define PICO_FLASH_SPI_CLKDIV 2
+#endif
+
+#ifndef PICO_FLASH_SIZE_BYTES
+#define PICO_FLASH_SIZE_BYTES (2 * 1024 * 1024)
```
Comment (kilograham) [inline src/boards/include/boards/seeed_xiao_rp2040.h:92]: based on discussion above, presumably this is wrong

Comment (ryanplusplus) [reply src/boards/include/boards/seeed_xiao_rp2040.h:92]: Per the [wiki](https://wiki.seeedstudio.com/XIAO-RP2040/) it's 2 MB and GD25Q16 is a 2 MB flash. I think @mcer12's comment above was 16 mbit, not mbyte.

**Review by lurch — APPROVED** [review]
Looks fine to me

**Review by kilograham — APPROVED** [review]

**Comment by ryanplusplus** [comment]
Thanks!

**Pull request merged**